packages/lib/src/interactions/box.ts

```diff
@@ -32,18 +32,24 @@ class Box extends Box3 {
   }
 
   public expandBySize(width: number, height: number): this {
-    return this.expandByVector(new Vector3(width / 2, height / 2, 0));
+    const { size } = this;
+
+    // Prevent shrinking to "negative" size - i.e. make sure `this.isEmpty()` is never `true`
+    const w = width < 0 ? Math.max(width, -size.width) : width;
```

Comment:
I wonder if we could make this condition more explicit
```suggestion
    const w = width < 0 && width < -size.width ? -size.width : width;
```

Up to you

Reply:
Good call

Reply:
Hmm after reading it again, the duplication of `-size.width` feels a bit less clear to me for some reason.

Reply:
Ideally, the condition should make clear that the final `width` is `0` but I couldn't find an easy way to do this since `width` and `height` are treated at the same time